src/Web/Static/documents/development/wips/5539-modularize-burn-searches.html.md

```diff
@@ -0,0 +1,152 @@
+        __in BURN_EXTENSION_MESSAGE message,
+        __in const LPVOID pvArgs,
+        __inout LPVOID pvResults,
+        __in_opt LPVOID pvContext
+        );
+
+    struct BURN_EXTENSION_CREATE_ARGS
+    {
+        DWORD cbSize;
+        DWORD64 qwEngineAPIVersion;
+        PFN_BURN_EXTENSION_ENGINE_PROC pfnBurnExtensionEngineProc;
+        LPVOID pvBurnExtensionEngineProcContext;
+    };
+
+    struct BURN_EXTENSION_CREATE_RESULTS
+    {
+        DWORD cbSize;
+        PFN_BURN_EXTENSION_PROC pfnBurnExtensionProc;
+        LPVOID pvBurnExtensionProcContext;
+    };
+
+    extern "C" typedef HRESULT(WINAPI *PFN_BURN_EXTENSION_CREATE)(
+        __in const BURN_EXTENSION_CREATE_ARGS* pArgs,
+        __inout BURN_EXTENSION_CREATE_RESULTS* pResults
+        );
+
+    extern "C" typedef void (WINAPI *PFN_BURN_EXTENSION_DESTROY)();
+
+    extern "C" typedef HRESULT(WINAPI *PFN_BURN_EXTENSION_ENGINE_PROC)(
+        __in BURN_EXTENSION_ENGINE_MESSAGE message,
+        __in const LPVOID pvArgs,
+        __inout LPVOID pvResults,
+        __in_opt LPVOID pvContext
+        );
+
+    enum BURN_EXTENSION_ENGINE_MESSAGE
+    {
+        BURN_EXTENSION_ENGINE_MESSAGE_GETVARIABLENUMERIC,
+        BURN_EXTENSION_ENGINE_MESSAGE_GETVARIABLESTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_GETVARIABLEVERSION,
+        BURN_EXTENSION_ENGINE_MESSAGE_FORMATSTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_ESCAPESTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_EVALUATECONDITION,
+        BURN_EXTENSION_ENGINE_MESSAGE_LOG,
+        BURN_EXTENSION_ENGINE_MESSAGE_SETVARIABLENUMERIC,
+        BURN_EXTENSION_ENGINE_MESSAGE_SETVARIABLELITERALSTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_SETVARIABLESTRING,
+        BURN_EXTENSION_ENGINE_MESSAGE_SETVARIABLEVERSION,
+    };
+
+2) Create a new `BurnExtension` element for the Burn manifest xml.
+
+    <xs:element name="BurnExtension">
+        <xs:complexType>
+            <xs:attribute name="Id" type="xs:string" />
+            <xs:attribute name="EntryPayloadId" type="xs:string" />
+            <xs:choice minOccurs="0" maxOccurs="unbounded">
+                <xs:element ref="PayloadRef" />
+            </xs:choice>
+        </xs:complexType>
+    </xs:element>
+
+3) Create a new `ExtensionSearch` element for the Burn manifest xml.
+
+    <xs:element name="ExtensionSearch">
+        <xs:annotation>
+            <xs:documentation>
+                The inner XML will contain the information needed for that specific search.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:complexType>
+            <xs:attribute name="Id" type="xs:string" />
+            <xs:attribute name="Variable" type="xs:string" />
+            <xs:attribute name="Condition" type="xs:string" />
+            <xs:attribute name="ExtensionId" type="xs:string" />
+        </xs:complexType>
+    </xs:element>
+
+4) Move the `Util` extension's searches out of the Burn engine and into this new `ExtensionSearch` framework.
```

Comment:
I'd start by moving them to dutil; that would simplify creating MSI custom actions out of the searches, something I know has been requested but cannot find an issue for atm.

Reply:
Sounds like a good idea. I was more interested in outlining the Burn specific code than where the actual code will be moved.